src/transform/src/predicate_pushdown.rs

```diff
@@ -69,9 +69,7 @@ impl crate::Transform for PredicatePushdown {
         _: TransformArgs,
     ) -> Result<(), crate::TransformError> {
         let mut empty = HashMap::new();
-        relation.visit_mut_pre(&mut |e| {
-            self.action(e, &mut empty);
-        });
+        self.action(relation, &mut empty);
```

Comment:
Does this change mean that we're only applying `PredicatePushdown` to certain parts of the `RelationExpr` now?

Reply:
Unless I've screwed something up, `action` should now be recursive, and will call itself as it descends the expression.

Reply:
Ideally we apply to transform to all parts of the dataflow graph, except that we want to avoid applying the general logic to the `Get` inside a `Filter`, as generally when we see a `Get` we want to say "whoa, no predicates should be pushed down!".